File: MemeShareJetpack/app/src/main/java/com/example/memesharejetpack/view/RegisterScreen.kt
package com.example.memesharejetpack.view

import android.widget.Toast
import androidx.compose.foundation.layout.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.runtime.livedata.observeAsState
import androidx.compose.runtime.saveable.rememberSaveable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.text.input.PasswordVisualTransformation
import androidx.compose.ui.unit.dp
import androidx.lifecycle.viewmodel.compose.viewModel
import com.example.memesharejetpack.viewmodel.RegisterViewModel

@Composable
fun RegisterScreen(
    onNavigateLogin: () -> Unit,
    onRegisterSuccess: () -> Unit,
    viewModel: RegisterViewModel = viewModel()
) {
    val context = LocalContext.current
    var username by rememberSaveable { mutableStateOf("") }
    var password by rememberSaveable { mutableStateOf("") }
    val registerResult by viewModel.registerResult.observeAsState()

    LaunchedEffect(registerResult) {
        registerResult?.let { res ->
            if (res.success) {
                Toast.makeText(context, "Kayıt başarılı! Giriş yapabilirsiniz", Toast.LENGTH_SHORT).show()
                onRegisterSuccess()
            } else if (res.message.isNotBlank()) {
                Toast.makeText(context, "Hata: ${res.message}", Toast.LENGTH_SHORT).show()
            }
        }
    }

    Scaffold { padding ->
        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(padding)
                .padding(24.dp),
            verticalArrangement = Arrangement.Center,
            horizontalAlignment = Alignment.CenterHorizontally
        ) {
            Text("Kayıt Ol", style = MaterialTheme.typography.headlineMedium)
            Spacer(Modifier.height(24.dp))
            OutlinedTextField(
                value = username,
                onValueChange = { username = it },
                label = { Text("Kullanıcı adı") },
                singleLine = true,
                modifier = Modifier.fillMaxWidth()
            )
            Spacer(Modifier.height(12.dp))
            OutlinedTextField(
                value = password,
                onValueChange = { password = it },
                label = { Text("Şifre") },
                singleLine = true,
                visualTransformation = PasswordVisualTransformation(),
                modifier = Modifier.fillMaxWidth()
            )
            Spacer(Modifier.height(20.dp))
            Button(
                onClick = {
                    if (username.isNotBlank() && password.isNotBlank()) {
                        viewModel.registerUser(username, password)
                    } else {
                        Toast.makeText(context, "Tüm alanları doldurun", Toast.LENGTH_SHORT).show()
                    }
                },
                modifier = Modifier.fillMaxWidth()
            ) { Text("Kayıt Ol") }

            TextButton(onClick = onNavigateLogin) {
                Text("Zaten hesabın var mı? Giriş yap")
            }
        }
    }
}

File: meme_share_flutter/lib/view/register_screen.dart
import 'package:flutter/material.dart';
import 'package:provider/provider.dart';

import '../components/snackbar.dart';
import '../viewmodel/register_vm.dart';

class RegisterScreen extends StatefulWidget {
  final VoidCallback onNavigateLogin;
  final VoidCallback onRegisterSuccess;

  const RegisterScreen({
    super.key,
    required this.onNavigateLogin,
    required this.onRegisterSuccess,
  });

  @override
  State<RegisterScreen> createState() => _RegisterScreenState();
}

class _RegisterScreenState extends State<RegisterScreen> {
  final _username = TextEditingController();
  final _password = TextEditingController();

  @override
  void dispose() {
    _username.dispose();
    _password.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return Consumer<RegisterVM>(
      builder: (context, vm, _) {
        return Scaffold(
          body: Padding(
            padding: const EdgeInsets.all(24),
            child: Center(
              child: SingleChildScrollView(
                child: Column(
                  crossAxisAlignment: CrossAxisAlignment.stretch,
                  children: [
                    Text('Kayıt Ol', style: Theme.of(context).textTheme.headlineMedium),
                    const SizedBox(height: 24),
                    TextField(
                      controller: _username,
                      decoration: const InputDecoration(labelText: 'Kullanıcı adı'),
                    ),
                    const SizedBox(height: 12),
                    TextField(
                      controller: _password,
                      decoration: const InputDecoration(labelText: 'Şifre'),
                      obscureText: true,
                    ),
                    const SizedBox(height: 20),
                    ElevatedButton(
                      onPressed: vm.isLoading
                          ? null
                          : () async {
                        final u = _username.text.trim();
                        final p = _password.text.trim();
                        if (u.isEmpty || p.isEmpty) {
                          showSnack(context, 'Tüm alanları doldurun');
                          return;
                        }

                        await vm.registerUser(u, p);
                        final res = vm.registerResult;

                        if (!mounted) return;

                        if (res?.success == true) {
                          showSnack(context, 'Kayıt başarılı! Giriş yapabilirsiniz');
                          widget.onRegisterSuccess();
                        } else {
                          showSnack(context, 'Hata: ${res?.message ?? vm.error ?? "Sunucu hatası"}');
                        }
                      },
                      child: vm.isLoading
                          ? const SizedBox(height: 18, width: 18, child: CircularProgressIndicator())
                          : const Text('Kayıt Ol'),
                    ),
                    TextButton(
                      onPressed: vm.isLoading ? null : widget.onNavigateLogin,
                      child: const Text('Zaten hesabın var mı? Giriş yap'),
                    ),
                  ],
                ),
              ),
            ),
          ),
        );
      },
    );
  }
}
